lib/vm/src/global.rs

```diff
@@ -63,13 +63,57 @@ impl Global {
         unsafe { NonNull::new_unchecked(ptr) }
     }
 
-    /// Get a reference to the definition
-    pub fn get(&self) -> &VMGlobalDefinition {
-        unsafe { &*self.vm_global_definition.get() }
+    /// Get a value from the global.
+    pub fn get<T>(&self) -> Value<T> {
+        let _global_guard = self.lock.lock().unwrap();
+        unsafe {
+            let definition = &*self.vm_global_definition.get();
+            match self.ty().ty {
+                Type::I32 => Value::from(*definition.as_i32()),
+                Type::I64 => Value::from(*definition.as_i64()),
+                Type::F32 => Value::F32(*definition.as_f32()),
+                Type::F64 => Value::F64(*definition.as_f64()),
+                Type::V128 => Value::V128(*definition.as_u128()),
```

Comment:
The fact two of these use From and the other two use enumerator constructors bothers me a little. Any reason for the inconsistency?

Reply:
didn't notice that! Good catch

Reply:
If using `from` always works, we could move that call outside the match.

Reply:
Good thought but each branch has to return the same value for that to work! So I don't think it'd be possible to structure this so that that worked

Reply:
I mean you could probably do `&dyn Into<Value>` as the type that they all share, but it's a little tricky to make that work